### PR TITLE
Timezone fix

### PR DESCRIPTION
**Pull request by btrott18**
Probably do not merge this yet, but want to get the ball rolling on the PR, so I can take care of any comments/suggestions while continuing to work through the code.

Biggest change so far was refactoring sendQueued.js

### CODE REVIEW

File: sendQueued.js [hunk 1]
```diff
@@ -1,130 +1,158 @@
...
-            knex('queued')
-              .where('queued_id', '=', queuedCitation.queued_id)
-              .update({'sent': true})
-              .exec(function(err, results) {
-                if (err) {
-                  console.log(err);
-                }
-              });
+      Promise.resolve()
+        .then(sendMessage(phone, process.env.TWILIO_PHONE_NUMBER, messages.greetingOneOfTwo()))
```
Comment (NigelKibodeaux) [inline sendQueued.js]: Last night, we changed this to only send one message. Twilio will handle the pagination for us.

File: sendQueued.js [hunk 2]
```diff
@@ -1,130 +1,158 @@
...
   return new Promise(function(resolve, reject) {
-    findQueued().then(function(resp) {
-      sendQueuedMessage(resp).then(resolve, reject);
-    }).catch(reject);
+    try{
+      findQueued()
+        .then(forEachResult(retrieveCitation))
+        .then(forEachResult(processCitationMessage))
+        .then(resolve, reject);
+    } catch(e) {
```
Comment (NigelKibodeaux) [inline sendQueued.js]: I'm a promise novice, but won't this promise pass errors to `.catch()` instead of throwing them?

File: utils/messages.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+
+	/**
+	 * Greeting message 1/2 template
+	 * 
+	 * @return {String} First of two greeting messages.
+	 */
+	greetingOneOfTwo: function() {
```
Comment (NigelKibodeaux) [inline utils/messages.js]: Like I said above, this was able to be consolidated into one message.

File: db.js [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
-var Knex = require('knex');
 require('dotenv').config();
-var knex = Knex.initialize({
+var knex = require('knex')({
   client: 'pg',
   connection: process.env.DATABASE_URL,
   pool: {
     afterCreate: function(connection, callback) {
-      connection.query("SET TIME ZONE 'America/Anchorage';", function(err) {
+      connection.query("SET TIME ZONE 'UTC';", function(err) {
```
Comment (NigelKibodeaux) [inline db.js]: Huh. I never would have thought we set a timezone in the database.

File: sendQueued.js (hunk 1 above)
Comment (btrott18) [inline sendQueued.js]: Updated to use a single message as well.

File: sendQueued.js (hunk 2 above)
Comment (btrott18) [inline sendQueued.js]: You're right, this pattern is frowned upon.  Changed to use ```.catch(reject)``` instead.

File: utils/messages.js (hunk 3 above)
Comment (btrott18) [inline utils/messages.js]: Updated.

**Comment by NigelKibodeaux** [comment]
@brendanbabb and @tescher, this PR looks good to me but I'm hesitant to merge it to master because I'm not familiar with how to test it. Does one of you want to do that so you can confirm that it works?

**Comment by btrott18** [comment]
Had a couple of dependencies goofed up...

Added logic to ensure tables exist - probably overkill, but ensures unit tests will run, and devs no longer have to execute ```npm run dbsetup```

**Comment by NigelKibodeaux** [comment]
@tescher this looks good to me. Do you want to give it a review and merge it if you agree?

Nice work, @btrott18 !

**Comment by tescher** [comment]
I can take a look tomorrow if that's OK.

Tim Escher
tim@eschers.com
608 432 5799

**Comment by tescher** [comment]
The code looks good to me (I agree, nice job cleaning it up). All tests
pass on my VM, but I would have a tough time testing this specific case
without putting it up on the test deployment *and* if there was data with
court dates early or late in the day that might trigger a bug.

How about:

1) Create a couple test cases with court times that fall early and late in
the day, and make sure the reminder job picks up the right ones.
2) Have @btrott18 and/or @NigelKibodeaux do a quick overview at the next
Hacker Night of Promises for the rest of us so we can understand what is
going on :)

As far as merging, I suggest doing it together at hacker night and
deploying it then so we can fix stuff if needed.

Tim Escher
tim@eschers.com
608-432-5799

**Comment by NigelKibodeaux** [comment]
That sounds like a good idea to me.

**Comment by btrott18** [comment]
Works for me

**Comment by tescher** [comment]
I think there may still be an issue here with the hard-coded "8 hours", which is the ANC offset (during DST no less):

var findReminders = function() {
   return knex('reminders')
     .where('sent', false)
     .join('cases', 'reminders.case_id', '=', 'cases.id')
     .whereRaw('("cases"."date" + interval \'**8 hours**\') < (now() + interval \'24 hours\')')
     .select();
 };

Also, can we have a couple tests written with court times right before and after midnight ANC time (like 11:30 PM and 12:30 AM) and make sure the reminders come out on the right day?

**Comment by btrott18** [comment]
I think that's a good idea.  I'll take a look this weekend.

**Comment by btrott18** [comment]
Hey guys

I won't be there tonight but wanted to give an update.  I am working
through the timezone stuff.  Think i have it fixed but have a bit more
cleanup to do...  There was a bit of a challenge around how the timestamp
was being stored.  Also I noticed it seems like we expect the timestamp to
have the time but then we store time in a separate field... Would it make
sense to just get rid of the time field and get time Frome the date
timestamp?

Thanks!
Brad

**Comment by tescher** [comment]
Thanks for your work on this Brad. If you can consolidate it down to one
field that's great. Reduces confusion in the future :)

Tim Escher
tim@eschers.com
608-432-5799

**Comment by btrott18** [comment]
So when the travis ci build runs, the database is storing the date at UTC offset 0, however when I run locally it is storing correctly, with UTC offset -08.  

Here is the log I am referring to:
RAW DATE FROM DB: 2015-03-03 23:00:00**+00**

On my local machine:
RAW DATE FROM DB: 2015-03-03 16:00:00**-08**

The same conversion was occurring locally when we were specifying the UTC timezone in the knex configuration.

As long as the database is not configured for UTC, and our date column is built with type timestamptz, then the date should be stored with the correct timezone offset.

Tim, do you know if there is database configuration somewhere I am not seeing?

**Comment by tescher** [comment]
Not sure. Maybe there is a environment variable in the .travis.yml that
needs to be set to set the TZ correctly in the db?

Time zones are a pain in the a**, aren't they? :)

Tim Escher
tim@eschers.com
608-432-5799

**Pull request merged**